### PR TITLE
Fix crash when drawing line gizmo with less than 2 vertices

### LINKED ISSUE #9089: Attempting to render empty linestrips causes a crash
## Bevy version

Bevy 0.11

## What you did

Have a system that unconditionally runs `gizmos.linestrip_2d(path, Color::YELLOW);`, where path is an iterator over a list of points that can be empty.

## What went wrong

It crashes with 
```
thread 'main' panicked at 'Buffer slices can not be empty', C:\Users\elabajaba\.cargo\registry\src\index.crates.io-6f17d22bba15001f\wgpu-0.16.2\src\lib.rs:2362:39
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Encountered a panic in exclusive system `bevy_render::renderer::render_system`!
```

## Additional information

I tested it on both vulkan and dx12.

It used to work on bevy main. https://github.com/bevyengine/bevy/pull/8910 is when it started crashing.

### PR DESCRIPTION
**Pull request by devil-ira**
# Objective

Fix #9089

## Solution

Don't try to draw lines with less than 2 vertices. These would not be visible either way.

### CODE REVIEW

**Comment by nicopap** [comment]
It should return success no?

**Comment by devil-ira** [comment]
I'm not sure what the semantics of `RenderCommandResult` are intended to be.
The `RenderCommandResult` doesn't appear to be used anywhere.

**Review by Selene-Amanita — APPROVED** [review]

**Review by nicopap — APPROVED** [review]
I'll approve for the sake of getting this merged quickly, but I would prefer if it was returning `RenderCommandResult::Success`

**Review by mockersf — COMMENTED** [review]
`RenderCommandResult` are currently ignored https://github.com/bevyengine/bevy/blob/eb485b1acc619baaae88d5daca0a311b95886281/crates/bevy_render/src/render_phase/draw.rs#L270-L271

But it would make much more sense for me too to return a `Success` here.

Also, surprisingly, the solution is not what this PR is actually doing. if you add 3 empty lines, the vertex count will be 3 and not hit that condition, still "rendering" empty lines

File: crates/bevy_gizmos/src/lib.rs [hunk 1]
```diff
@@ -462,6 +462,10 @@ impl<P: PhaseItem> RenderCommand<P> for DrawLineGizmo {
             return RenderCommandResult::Failure;
         };
 
+        if line_gizmo.vertex_count < 2 {
+            return RenderCommandResult::Failure;
```
Comment (mockersf) [inline crates/bevy_gizmos/src/lib.rs:466]:
```suggestion
            return RenderCommandResult::Success;
```

**Comment by mockersf** [comment]
adding a gizmo line adds the line position to the list, then unconditionally push an "empty" slot with `NAN`. could we instead not push those nans when we didn't push actual data to the lines?

**Comment by nicopap** [comment]
Also I learned recently that we should prefer u32::MAX over NaN. It is the "official" way of telling the GPU that our line list is ending.

**Comment by alice-i-cecile** [comment]
@devil-ira can you make the `RenderCommandResult` change requested and then I'll merge this in for you.

**Comment by mockersf** [comment]
> @devil-ira can you make the `RenderCommandResult` change requested and then I'll merge this in for you.

I would prefer if someone tries to fix the issue at the source rather than this PR. We are pushing garbage in the mesh, and this PR just says "oh if there's only one garbage, don't draw it"

**Comment by devil-ira** [comment]
When we add a retained version of the line gizmo well need this check either way as users could easily create a linestrip of length 0 or 1.

**Comment by alice-i-cecile** [comment]
@mockersf @superdump; I'll defer to one of you on whether we merge this PR as is.

**Review by DGriffin91 — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
I'm going to merge this following the approval from @DGriffin91, whose expertise I trust here: crashes are very sever bad and we can revert this very simple fix in a PR with a better fix later.

**Review by superdump — APPROVED** [review]

**Comment by DGriffin91** [comment]
@alice-i-cecile imo we should try to get this into [0.11.1](https://github.com/bevyengine/bevy/milestone/15) if we're merging it now. I noticed it was removed from the milestone, not sure if that means it would no longer be cherry picked for the point release.

**Comment by alice-i-cecile** [comment]
I'll defer to @cart for that decision, but I do agree.

**Comment by cart** [comment]
Sorry I didn't check my notifications before publishing so I missed this.

**Pull request merged**